example/lib/ide/problems_panel/diagnostics.dart

```diff
@@ -0,0 +1,59 @@
+class Diagnostic {
+  final String code;
+  final String message;
+  final Range range;
+  final int severity;
+  final String source;
+
+  Diagnostic({
+    required this.code,
+    required this.message,
+    required this.range,
+    required this.severity,
+    required this.source,
+  });
+
+  factory Diagnostic.fromJson(Map<String, dynamic> json) {
+    return Diagnostic(
+      code: json['code'] as String,
+      message: json['message'] as String,
+      range: Range.fromJson(json['range'] as Map<String, dynamic>),
+      severity: json['severity'] as int,
+      source: json['source'] as String,
+    );
+  }
+}
+
+class Range {
+  final Position start;
+  final Position end;
+
+  Range({
+    required this.start,
+    required this.end,
+  });
+
+  factory Range.fromJson(Map<String, dynamic> json) {
+    return Range(
+      start: Position.fromJson(json['start'] as Map<String, dynamic>),
+      end: Position.fromJson(json['end'] as Map<String, dynamic>),
+    );
+  }
+}
+
+class Position {
```

Comment:
I think we already have `Range` and `Position` defined in `example/lib/lsp_exploration/lsp/messages/common_types.dart`